db/record.c

```diff
@@ -917,6 +929,18 @@ int upd_record(struct ireq *iq, void *trans, void *primkey, int rrn,
         goto err;
     }
 
+    /* Also check blob sizes */
+    if (!(flags & RECFLAGS_NO_BLOBS)) {
+        for (int i = 0; i < maxblobs; i++) {
+            if (blobs[i].exists && blobs[i].length > MAXBLOBLENGTH) {
+                reqerrstr(iq, COMDB2_ADD_RC_INVL_BLOB,
+                        "blob size (%d) exceeds maximum (%d)", blobs[i].length, MAXBLOBLENGTH);
+                retrc = ERR_BLOB_TOO_LARGE;
+                ERR;
+            }
+        }
+    }
+
```

Comment:
Not sure that'll make it clearer.  If anything, that code should be folded into check_blob_consistency